server.js

```diff
@@ -20,25 +20,51 @@ app.get('/rooms', (req, res) => {
 
 io.on('connection', (socket) => {
     console.log('Client Connected')
-
-    socket.on('chat', (message) => {
-        const room = Object.keys(socket.rooms)
-            .filter(x => x !== socket.id)
-
-        io.to(room[0]).emit('chat', message)
+    
+	socket.on('chat', (message) => {
+        notifyRoom(socket, 'chat', message)
         console.log(message)
     })
 
     socket.on('private', (sender, message) => {
         // private message received
     })
-
-    socket.on('disconnect', () => {
-        io.emit('client disconnected')
+    
+    socket.on('leave-room', (name) => {
+        const room = filterRooms(socket)
+        const message = `${getTime()} - ${name} has disconnected`
+        notifyRoom(room, 'leave-room', message)
+		console.log(`${message} from ${room}`) 
+ 
+        socket.leave(room, (err) => {
+            if (err) {
+                console.log(err)
+            }
+        })
     })
 
-    socket.on('create', (room) => {
+    socket.on('join-room', (room, name) => {
         socket.join(room)
+        const message =  `${getTime()} - ${name} has connected`
+        notifyRoom(room, 'join-room', message)
+        console.log(`${message} to ${room}`)
     })
 })
 
+function notifyRoom(room, event, message) {
+    if (typeof(room) !== 'string') {
+        // if the socket is passed
+        room = filterRooms(room)
+    }
+    io.to(room).emit(event, message)
+}
+
+function filterRooms(socket) { 
+    return Object.keys(socket.rooms)
+        .filter(x => x !== socket.id)[0]
+} 
+
+function getTime() {
+	let d = new Date()
+	return `${(d.getHours() >= 12 ? d.getHours() - 12 : d.getHours())}:${d.getMinutes()}`
```

Comment:
I thought we were going with 24h time?

Also if `getMinutes()` returns a number less than 10 the time will look like `22:9`